### ISSUE TITLE
3DX CATIA Mouse Controls

### ISSUE THREAD

**Issue by jgomez720** [KittyCAD/modeling-app#2134]
Sent to us from Nick Boone:

>**primary method**
Pan: Click and hold the scroll wheel, then move the cursor in any direction
Zoom: Hold Ctrl, then hold the scroll wheel, then move the cursor up or down
Rotate (about center): With the cursor in the center 1/3rd of the screen, hold the scroll wheel, then hold Ctrl, then move the cursor in any direction
Rotate (about camera): With the cursor towards the edge of the screen, hold the scroll wheel, then hold Ctrl, then move the cursor up or down

>**lesser used alternate method**
Pan: Click and hold the scroll wheel, then move the cursor in any direction
Zoom: Hold down the scroll wheel, then left or right click and release, then move cursor up or down
Rotate: Hold down the scroll wheel, then hold the left or right mouse button, then move the cursor in any direction.

>- Right click is just right click. You get a short menu with display options.
>- Scrolling the scroll wheel is surprisingly never used to move the part, with or without the Ctrl key. It can be used for navigating the feature tree though.

>These controls generally stay pretty consistent across all the different roles in the 3DX native app (assembly design, part design, electrical routing, etc). Some of the product explorer and change management tools have different methods of navigating, but they aren't used for much actual design work. 

>It's not intuitive at all - particularly having to remember whether to hit Ctrl or scroll wheel first for zoom vs rotate. The sequence of buttons matters for each movement. Admittedly though, it's extremely fast once you get the hang of it. There are plenty of superusers who prefer this control scheme to anything else. Personally, I don't love having to use both hands so frequently. It makes it difficult to work while eating a bagel. Dassault does offer a license which allows you to change the settings to Solidworks controls, but it's not included as a base feature.

**Comment by franknoirot** [comment]
First of all

> Dassault does offer a license which allows you to change the settings to Solidworks controls, but it's not included as a base feature.

WHAT

**Comment by franknoirot** [comment]
Also it seems wild to me to go into different movement types depending on whether you click first or hit <kbd>Ctrl</kbd> first, am I reading that right @jgomez720?

**Comment by franknoirot** [comment]
We might be able to support that I'm just not sure As long as—for the time being—we limit the scope to cover just the primary method and not support orbit about camera (which is a pretty dramatic expansion to our orbit system). Then we can break the other parts of this out into a Discussion to see if we want to support them. 

@jgomez720 do you think we could implement a "CATIA-like" set of controls that use Ctrl+Shift+middle-click-drag for Orbit?

@jtran I'm gonna assign this to you if you don't mind, since it's not too far afield from your other initial issues. I think it should be trivial for you to add my suggested "CATIA-like" set of controls. I would consider this low priority compared to sketch and modeling improvements.

**Comment by jgomez720** [comment]
Yeah this is a low priority to our other issues. I'd say if it's something quick we could implement with the CATIA-like controls, go for it, but if it would take some time, put it aside for now. The 1/3rd screen thing that he's mentioning too was true in NX too, but imo it's not needed to get your desired result. 

We give customers what they must have first, worry about what's nice to have after v1
